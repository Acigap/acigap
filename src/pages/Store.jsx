import React from 'react';

function Store() {
  const products = [
    {
      id: 1,
      name: "Bambu Lab P1P",
      price: 59900,
      image: "https://f.btwcdn.com/store-34999/product-thumb/11e7bc6c-2bd1-5844-a050-6666ba0ec7d3.jpg",
      description: "Professional 3D printer with CoreXY system and high speed printing",
      category: "Featured"
    },
    {
      id: 2,
      name: "Bambu Lab X1 Carbon",
      price: 89900,
      image: "https://f.btwcdn.com/store-34999/product-thumb/5d847bed-3dde-f6c9-135a-6666bbd6b058.jpg",
      description: "High-end 3D printer with carbon fiber composite frame",
      category: "Featured"
    },
    {
      id: 3,
      name: "Bambu Lab A1 Mini",
      price: 29900,
      image: "https://f.btwcdn.com/store-34999/product-thumb/4ca61bd9-6bc4-2d2d-8aa5-6819e61dbbd2.jpg",
      description: "Compact and affordable 3D printer for beginners",
      category: "New"
    },
  ];

  const categories = ["All", "Featured", "New"];
  const [selectedCategory, setSelectedCategory] = React.useState("All");

  const filteredProducts = selectedCategory === "All" 
    ? products 
    : products.filter(product => product.category === selectedCategory);

  return (
    <div className="space-y-8">
      <div className="text-center space-y-4">
        <h1 className="text-4xl font-bold text-[var(--neutral-12)]">Our Products</h1>
        <p className="text-[var(--neutral-11)] max-w-2xl mx-auto">
          Discover our range of high-quality products designed to meet your needs
        </p>
      </div>

      {/* Category Filter */}
      <div className="flex justify-center space-x-4">
        {categories.map(category => (
          <button
            key={category}
            onClick={() => setSelectedCategory(category)}
            className={`px-4 py-2 rounded-full text-sm font-medium transition-colors
              ${selectedCategory === category 
                ? 'bg-[var(--primary-green)] text-white' 
                : 'bg-[var(--neutral-3)] text-[var(--neutral-11)] hover:bg-[var(--neutral-4)]'
              }`}
          >
            {category}
          </button>
        ))}
      </div>

      {/* Product Grid */}
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8">
        {filteredProducts.map(product => (
          <div 
            key={product.id}
            className="group bg-[var(--neutral-1)] rounded-lg overflow-hidden border border-[var(--border)] hover:border-[var(--primary-green)] transition-colors"
          >
            <div className="aspect-video relative overflow-hidden">
              <img 
                src={product.image} 
                alt={product.name}
                onError={e => { e.target.onerror = null; e.target.src = "/logo192.png"; }}
                className="object-cover w-full h-full group-hover:scale-105 transition-transform duration-300"
              />
              {product.category === "New" && (
                <div className="absolute top-4 right-4 bg-[var(--primary-green)] text-white px-3 py-1 rounded-full text-sm font-medium">
                  New
                </div>
              )}
            </div>
            <div className="p-6 space-y-4">
              <h3 className="text-xl font-semibold text-[var(--neutral-12)]">{product.name}</h3>
              <p className="text-[var(--neutral-11)]">{product.description}</p>
              <div className="flex justify-between items-center">
                <p className="text-2xl font-bold text-[var(--neutral-12)]">
                  à¸¿{product.price.toLocaleString()}
                </p>
                <button className="bg-[var(--primary-green)] text-white px-6 py-2 rounded-lg hover:bg-[var(--primary-green-hover)] transition-colors">
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}

export default Store;