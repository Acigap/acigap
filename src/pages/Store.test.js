import React from 'react';
import { render } from '@testing-library/react';
import Store from './Store';

describe('Store', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Store />);
    expect(asFragment()).toMatchSnapshot();
  });
});
