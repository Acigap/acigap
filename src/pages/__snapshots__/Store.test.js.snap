// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="space-y-8"
  >
    <div
      class="text-center space-y-4"
    >
      <h1
        class="text-4xl font-bold text-[var(--neutral-12)]"
      >
        Our Products
      </h1>
      <p
        class="text-[var(--neutral-11)] max-w-2xl mx-auto"
      >
        Discover our range of high-quality products designed to meet your needs
      </p>
    </div>
    <div
      class="flex justify-center space-x-4"
    >
      <button
        class="px-4 py-2 rounded-full text-sm font-medium transition-colors
              bg-[var(--primary-green)] text-white"
      >
        All
      </button>
      <button
        class="px-4 py-2 rounded-full text-sm font-medium transition-colors
              bg-[var(--neutral-3)] text-[var(--neutral-11)] hover:bg-[var(--neutral-4)]"
      >
        Featured
      </button>
      <button
        class="px-4 py-2 rounded-full text-sm font-medium transition-colors
              bg-[var(--neutral-3)] text-[var(--neutral-11)] hover:bg-[var(--neutral-4)]"
      >
        New
      </button>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8"
    >
      <div
        class="group bg-[var(--neutral-1)] rounded-lg overflow-hidden border border-[var(--border)] hover:border-[var(--primary-green)] transition-colors"
      >
        <div
          class="aspect-video relative overflow-hidden"
        >
          <img
            alt="Bambu Lab P1P"
            class="object-cover w-full h-full group-hover:scale-105 transition-transform duration-300"
            src="https://f.btwcdn.com/store-34999/product-thumb/11e7bc6c-2bd1-5844-a050-6666ba0ec7d3.jpg"
          />
        </div>
        <div
          class="p-6 space-y-4"
        >
          <h3
            class="text-xl font-semibold text-[var(--neutral-12)]"
          >
            Bambu Lab P1P
          </h3>
          <p
            class="text-[var(--neutral-11)]"
          >
            Professional 3D printer with CoreXY system and high speed printing
          </p>
          <div
            class="flex justify-between items-center"
          >
            <p
              class="text-2xl font-bold text-[var(--neutral-12)]"
            >
              ฿59,900
            </p>
            <button
              class="bg-[var(--primary-green)] text-white px-6 py-2 rounded-lg hover:bg-[var(--primary-green-hover)] transition-colors"
            >
              Add to Cart
            </button>
          </div>
        </div>
      </div>
      <div
        class="group bg-[var(--neutral-1)] rounded-lg overflow-hidden border border-[var(--border)] hover:border-[var(--primary-green)] transition-colors"
      >
        <div
          class="aspect-video relative overflow-hidden"
        >
          <img
            alt="Bambu Lab X1 Carbon"
            class="object-cover w-full h-full group-hover:scale-105 transition-transform duration-300"
            src="https://f.btwcdn.com/store-34999/product-thumb/5d847bed-3dde-f6c9-135a-6666bbd6b058.jpg"
          />
        </div>
        <div
          class="p-6 space-y-4"
        >
          <h3
            class="text-xl font-semibold text-[var(--neutral-12)]"
          >
            Bambu Lab X1 Carbon
          </h3>
          <p
            class="text-[var(--neutral-11)]"
          >
            High-end 3D printer with carbon fiber composite frame
          </p>
          <div
            class="flex justify-between items-center"
          >
            <p
              class="text-2xl font-bold text-[var(--neutral-12)]"
            >
              ฿89,900
            </p>
            <button
              class="bg-[var(--primary-green)] text-white px-6 py-2 rounded-lg hover:bg-[var(--primary-green-hover)] transition-colors"
            >
              Add to Cart
            </button>
          </div>
        </div>
      </div>
      <div
        class="group bg-[var(--neutral-1)] rounded-lg overflow-hidden border border-[var(--border)] hover:border-[var(--primary-green)] transition-colors"
      >
        <div
          class="aspect-video relative overflow-hidden"
        >
          <img
            alt="Bambu Lab A1 Mini"
            class="object-cover w-full h-full group-hover:scale-105 transition-transform duration-300"
            src="https://f.btwcdn.com/store-34999/product-thumb/4ca61bd9-6bc4-2d2d-8aa5-6819e61dbbd2.jpg"
          />
          <div
            class="absolute top-4 right-4 bg-[var(--primary-green)] text-white px-3 py-1 rounded-full text-sm font-medium"
          >
            New
          </div>
        </div>
        <div
          class="p-6 space-y-4"
        >
          <h3
            class="text-xl font-semibold text-[var(--neutral-12)]"
          >
            Bambu Lab A1 Mini
          </h3>
          <p
            class="text-[var(--neutral-11)]"
          >
            Compact and affordable 3D printer for beginners
          </p>
          <div
            class="flex justify-between items-center"
          >
            <p
              class="text-2xl font-bold text-[var(--neutral-12)]"
            >
              ฿29,900
            </p>
            <button
              class="bg-[var(--primary-green)] text-white px-6 py-2 rounded-lg hover:bg-[var(--primary-green-hover)] transition-colors"
            >
              Add to Cart
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
